### ISSUE TITLE
SVGLength.m:571 Assert crash

### ISSUE THREAD

**Issue by mylogon341** [SVGKit/SVGKit#839]
### New Issue Checklist

* [X] I have read and understood the [CONTRIBUTING guide](https://github.com/SVGKit/SVGKit/blob/2.x/CONTRIBUTING.md)
* [ ] I have reproduced the problem while running in Xcode, and read the Console log, and acted upon any relevant error or warning messages
* [ ] I have reproduced the problem while running in Xcode's Debugger, and used breakpoints to examine the actual values of variables vs the expected values
* [X] I am not using CocoaPods, or: I am using CocoaPods, but have verified my podspec points to SVGKit v3.x

### Issue Description and Steps

Please fill in the detailed description of the issue (full output of any stack trace, compiler error, ...) and the steps to reproduce the issue.

So, this is something that I actually can't reproduce. A few months ago, I was getting quite a few crashes in production because there was a typo in the iPad 10th Gen model number, which I fixed with a PR here #826 

I marked the issue as resolved and got around to putting out a new build once the PR was signed off soon after. I'm now getting new crashes in as a regression issue for that exact same iPad model (along with other M4 models), despite those now being included in this dictionary.

![Screenshot 2024-08-05 at 10 26 48 AM](https://github.com/user-attachments/assets/f3299739-516b-4723-9c60-1961f4ef7f64)

I'm 99% sure my latest build (a few weeks ago) went out with the latest 3.x with these new values in, but I might be wrong. I would be surprised if Xcode wasn't pulling the latest 3.x on the CI machine though, but I will try to find a way to make sure it's forcing it.

2 questions:
1. Why might these included values be still causing issues in the the latest prod build?
2. More importantly, how can I stop these kind of things from crashing in production? I would much rather SVGKit fall back on a best guess, but I'm getting fatal errors when listed devices are missing. I cannot always get a new build out as soon as a new device is released, so would love to avoid this.

However, this is a bit of a strange one, because I cannot reproduce it. I have launched my production app on Browserstack with a 10th Gen iPad and am not seeing this fatal error. Is it possible that these failed NSAsserts are triggering something in firebase? I can't see that being the case, but I need to check. 

Thanks for any advice on this. 
Luke

**Comment by adamgit** [comment]
If you're getting assertions in live builds then your build process
(unrelated to SVGKit) is broken somewhere - Apple strips all NSAssert from
live/production builds (Xcode ships by default with 'DEBUG' and
'PRODUCTION' IIRC). By design it's impossible to see an Assertion on an
end-user device unless you've sent them a 'DEBUG' build.

In production builds, the code falls-back and gives a number - which will
cause an SVG that is resolution-dependent to render slightly larger or
smaller than expected, but otherwise as normal.

Net effect:
 - in production, users might see 'wrong-sized' images since there's no
OS-provided way for us to fetch the PPI, so until someone edits the code
there's no way for us to know the 'correct' size
 - in debug/development/local builds, you'll get crashes due to NSAssert --
as an in-your-face warning that unless you update the code, your app will
be slightly incorrect if you ship a production build now.

**Comment by mylogon341** [comment]
I'm really quite confused on this one then. I have verified that my archive method is definitely 'Production' config and has the `-Os` optimisation against it, which should strip out NSAsserts, afaik.
The only thing I cannot confirm, and don't really know much about, admittedly, is the optimisation level of the SPM package. I am importing SVGKit via SPM and don't know if Xcode is actually using that compiler flag against the packages too. 
If it's not, then that might explain why I'm seeing the NSAssert in production.
For the time being, I might try porting this over to a Cocoapods dependency to see if it makes any difference for the next release as im kind of grasping at straws on this one 🤔

**Comment by adamgit** [comment]
Yep, strange - I'd guess the SPM too, the toolchain there has always been a
bit behind/wobbly on Apple's side. Very disappointing though if true.

**Comment by pixelmatrix** [comment]
I have the same issue. NSAssert is crashing in App Store builds. I've integrated this package via SPM, and it looks like this is a known quirk. The Package.swift for this library should be updated to disable the assertions:

https://forums.swift.org/t/assertions-in-swift-packages/42692

**Comment by adamgit** [comment]
Wow, good find. Also: pretty embarassing that xcode needs this manual
correction :). Submit a PR?

**Comment by pixelmatrix** [comment]
Opened here: https://github.com/SVGKit/SVGKit/pull/842
